tools/generator/Utilities/TypeNameUtilities.cs

```diff
@@ -101,5 +102,41 @@ public static string StudlyCase (string name)
 			}
 			return builder.ToString ();
 		}
+
+		// From https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Utilities/ResourceIdentifier.cs
+		// https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+		private const string FormattingCharacter = @"\p{Cf}";
+		private const string ConnectingCharacter = @"\p{Pc}";
+		private const string DecimalDigitCharacter = @"\p{Nd}";
+		private const string CombiningCharacter = @"\p{Mn}\p{Mc}";
+		private const string LetterCharacter = @"\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}";
+
+		private const string IdentifierPartCharacter = LetterCharacter +
+		    DecimalDigitCharacter +
```

Comment:
This code is using spaces instead of tabs for indentation.  Please fix. :-)

(Which is odd, as *other methods* in this file are using tabs...)